### DISCUSSION TITLE
Outline - Feature Extraction

### DISCUSSION THREAD

**Discussion by johko** [huggingface/computer-vision-course#101] [01. Fundamentals]
Here is the outline for the Feature Extraction section:

- [ ] **Feature Detection** - feature_description.mdx - @blessondavis

  - What are features?
  - What makes a good feature?
  - Simple Feature Detection (e.g. Harris Corner Detection, Hough Transform, Shi-Tomasi)

- [ ] **Feature Description** - feature_description.mdx - @aman06012003
  - How can we represent features in data structures? 
  - What makes a good descriptor (think of different image conditions, rotation, translation, etc.)
  - Example algorithm: SIFT & SURF

- [ ] **Feature Matching** - feature_matching.mdx - @kfahn22
  - How can we match detected features from image-to-image? 
  - Examples (Brute Force, FLANN, maybe others)

- [ ] **CNN Features + Visualization** -  feature_visualization.ipynb - @johko 
  - How do ANN (especially CNNs) detect features?
  - What do the features “look like”?
  - Some nice visualizations, maybe some DeepDream for fun

- [ ] **Feature Extraction Notebook** - feature_extraction.ipynb - @BSophia 

- [ ] **Feature Visualization Notebook** - feature_visualization.ipynb - @JavidanAbdullayev

Resources:
https://docs.opencv.org/4.8.0/db/d27/tutorial_py_table_of_contents_feature2d.html
https://towardsdatascience.com/image-feature-extraction-traditional-and-deep-learning-techniques-ccc059195d04
https://fpcv.cs.columbia.edu/

Working on this part are @kfahn22, @aman06012003, @blessondavis, @JavidanAbdullayev, @BSophia, @johko

**Comment by kfahn22** [comment]
I am trying to trouble-shoot something that showed up in the quality test.  I didn't add a link to an actual image in feature-matching.mdx and I am guessing this is the issue. (At least the test failed at line 27):
img1 = cv.imread('image1.png',cv.IMREAD_GRAYSCALE)

Nevermind, I have fixed the issue.
